src/cls/IPM/Utils/Migration.cls

```diff
@@ -0,0 +1,209 @@
+Include %occInclude
+
+Class %IPM.Utils.Migration
+{
+
+ClassMethod RunAll(verbose As %Boolean = 1) As %Status
+{
+    Set sc = $$$OK
+    Try {
+        Do ..MigrateZPMToIPM(verbose)
+    } Catch e {
+        Set sc = e.AsStatus()
+    }
+    Quit sc
+}
+
+ClassMethod MigrateZPMToIPM(verbose As %Boolean = 1)
+{
+    Set oldTopPackage = "%ZPM.PackageManager."
+    If $Order(^oddCOM(oldTopPackage)) '[ oldTopPackage {
+        Write:verbose !,"Older IPM version not found; nothing to migrate.",!
+        Quit
+    }
+
+    Set sc = $$$OK
+    Set successPackageList = ""
+    Set failedPackageList = ""
+
+    // Intentionally runs for current namespace only
+    Try {
+        // ^ZPM.Dev.ModuleD -> ^IPM.Storage.ModuleD
+        For lockName = $Name(^ZPM.Dev.ModuleD),$Name(^IPM.Storage.ModuleD) {
+            Set locks(lockName) = $System.AutoLock.Lock(lockName,,2)
```

Comment:
@isc-kiyer I think I might have written LockManager (at least I've written things like it before) - it may be cleaner to manage locks in a single object rather than a bunch of objects, so I wouldn't necessarily throw it out just yet. Although perhaps it could be updated to use AutoLock under the hood.